stdpopsim/slim_engine.py

```diff
@@ -599,6 +599,47 @@
 
 """
 
+_raw_stdpopsim_top_level_schema = {
```

Comment:
Can you give this schema a quick look @petrelharp ?

Reply:
Let's also stick in an `id`, since [all DFEs have IDs](https://github.com/popsim-consortium/stdpopsim/blob/main/stdpopsim/dfe.py#L176) and something we'd really like to get out of the metadata is which mutation type goes with which DFE?

Reply:
And - how about the mutation type entries look exactly like what we get if we do `mut_type.asdict()`?

Reply:
yes, good call! maybe also add the description to the DFE?

there is no `asdict` method to `MutationType`. But the schema is basically the same as `mut_type.__dict__` with three more properties? `slim_mutation_type_id`, `is_neutral`, `simulated_in_SLiM` which are needed for Xin's analyses.

Reply:
note that the `mutation_types` are nested within each DFE in this schema, so you can tell which mt is with each DFE.